### DISCUSSION TITLE
M1 mac (ARM) build fails as 'damaged'

### DISCUSSION THREAD

**Discussion by entangledothers** [cables-gl/cables_docs#284] [! Welcome - readme first!]
There seems to be an issue with the ARM builds produced by the building to executable function, they consistently fail with the error message `"patch.app" is damaged and can't be opened. You should move it to the Bin.`: 

<img width="372" alt="Screenshot 2022-06-26 at 11 33 58" src="https://user-images.githubusercontent.com/5319246/175810153-2344abd3-63df-41b7-a7d4-cc4d520dc81b.png">

Running the same build in its intel variety works just fine via Rosetta. This is with Mac OS 12.4 and an M1 mac. Furthermore this isn't seemingly a security issue, the intel build requires the override via system preferences to run, but the ARM build fails before even requiring that step in order to run. Same build runs perfectly on windows.

**Comment by steam0r** [comment]
hey,

export to m1 and running the arm64 build runs fine for me, it really seems to be dependend on the system settings and
wether or not you have allowed "things like this" to run before. could you test any/some/all of these and report back
what helped, so we can at least add some steps to mitigrate this to the documentation, or create a build that does
this automatically on fist run...thanks:

https://appuals.com/damaged-app-cannot-be-opened-on-macos-error/

best,
stephan

**Reply by entangledothers** [reply]
Hey and thanks for the prompt followup!
Here's the list of steps and their results as tested:

* **Method 1:** Works! 
* **Method 2:** Doesn't work. 
* **Method 3:** Doesn't seem to be an option any longer in 12.4, can only choose between 'app store' and 'app store and identified developers'. I.e. doesn't work.
* **Method 4:** Doesn't work. 

So, thankfully method 1 works, but it's a bit tricky as it requires that being done on whichever machine it is being run as well as requiring the person opening to have admin rights, is there any option to automate this step so as to make it more intuitive for the end user/viewer/something? 

Thanks so much!

**Reply by steam0r** [reply]
i will look into this further, i THINK the admin rights will be needed at least once, otherwise we'd have to sign the app somehow. which then would not work with building this on your github...so me might need to move this to our servers. but i'll have to check first if not a small "runfirsttime.sh" might be okay for now as well...

have you tried enabling the security checks again and then running the patch again? does that work?

**Reply by steam0r** [reply]
ah, sorry, confused, method 2 is disabling checks, ignore me then

**Reply by mediajolt001** [reply]
On Sonoma, this no longer seems to work - at least for me, on an M1 MacBook Pro. It doesn't even see it as an app, the icon has the circle/slash symbol for both ARM and Intel builds.

**Reply by steam0r** [reply]
for fresh exports? just tested this and i dont have the circle/slash symbol. 

you are right that apple seems to get worse in regards to non-signed apps, and now `xattr -cr` seems to be the only way to get an export to run on m1 with recent operating systems...nice...
